src/Compiler/TypedTree/TcGlobals.fs

```diff
@@ -1852,6 +1851,10 @@ type TcGlobals(
 
   member _.DebuggerNonUserCodeAttribute = debuggerNonUserCodeAttribute
 
+  member _.HasTailCallAttrib (attribs: Attribs) =
+    attribs
+    |> List.exists (fun a -> a.TyconRef.CompiledRepresentationForNamedType.FullName = "Microsoft.FSharp.Core.TailCallAttribute")
+  
```

Comment:
Can't we use something like findPublicSysAttrib or tryFindSysAttrib? Like we do with RequiredAttribute for example. One of them _should_ be able to find them in all ccus

Reply:
I tried the following:
```fsharp
  member _.HasTailCallAttrib (attribs: Attribs) =
      let tailCallAttr = findSysAttrib "Microsoft.FSharp.Core.TailCallAttribute"
      List.exists
              (fun (Attrib(tyconRef = tcref2)) ->
                  tcref2.CompiledRepresentationForNamedType.FullName = tailCallAttr.TyconRef.CompiledRepresentationForNamedType.FullName)
              attribs
```
But that raises the following:
```
FSC : error FS0074: The type referenced through 'Microsoft.FSharp.Core.TailCallAttribute' is defined in an assembly that is not referenced. You must add a reference to assembly 'The system type 'Microsoft.FSharp.Core.TailCallAttribute' was required but no referenced system DLL contained this type'. [C:\Users\schae\src\tailcallaltat 
tr\tailcallaltattr.fsproj]
```
As `tryFindSysAttrib` also uses `tryFindSysTypeCcuHelper` internally, that doesn't work either.

To better illustrate the usecase, here's a [repo](https://github.com/dawedawe/tailcallaltattr), that uses FSharp.Core 6.0.7 and gets TailCall warnings with this PR.

Reply:
It bothers me slightly that we just do string comparison in the list of attributes, instead of checking as we usually do. 

It's not the hill I'm willing to die on though, if everyone else is fine with it.

Reply:
Thanks for the reviews so far.
But, uhm, what's the state here: Can we go on with this or are changes required?
If the string comparison is not okay, can you explain what could go wrong?

Reply:
Two things - we don't really do it usually, and by checking the name we don't actually check it's a proper attribute (as attribute functions ensure).

That said, I'm fine with merging it, I can fix it later.